Clean up 8 `any`-type lint warnings (batch.test.ts + ai-helpers.ts)

## Why
\`pnpm lint\` is at 0 errors but 8 warnings, all of the same kind: \`@typescript-eslint/no-explicit-any\`. The codebase has a \"0 errors policy\" but warnings are accumulating — this is a small, well-scoped cleanup that's a great first contribution.

## State (2026-04-29)

\`pnpm lint\` output:

\`\`\`
packages/cli/src/commands/ai-helpers.ts
   37:63  warning  Unexpected any. Specify a different type

packages/cli/src/commands/batch.test.ts
    63:46  warning  Unexpected any. Specify a different type
    94:41  warning  Unexpected any. Specify a different type
   150:41  warning  Unexpected any. Specify a different type
   161:45  warning  Unexpected any. Specify a different type
   209:41  warning  Unexpected any. Specify a different type
   256:41  warning  Unexpected any. Specify a different type
   336:29  warning  Unexpected any. Specify a different type

✖ 8 problems (0 errors, 8 warnings)
\`\`\`

## Scope

- [ ] Read each call site and replace \`any\` with the correct type
- [ ] For \`batch.test.ts\`: most of these are likely test fixture / mock typings — \`unknown\` + a runtime assertion is often the right call
- [ ] For \`ai-helpers.ts\`: real production code, take the time to thread the actual type
- [ ] After: \`pnpm lint\` should report \`✔ No ESLint warnings or errors\` for the CLI package
- [ ] No new \`eslint-disable\` comments — fix the types properly

## How to verify

\`\`\`bash
pnpm install && pnpm build
pnpm -F @vibeframe/cli lint    # should show 0 warnings
pnpm -F @vibeframe/cli test    # tests should still pass
\`\`\`

## Notes for first-timers

- The repo uses TypeScript strict mode and ESM throughout
- Run \`pnpm -r exec tsc --noEmit\` after your edits — \`pnpm build\` uses esbuild bundling and skips strict typecheck (see CLAUDE.md)
- See [CONTRIBUTING.md](../CONTRIBUTING.md) for the PR flow

--- a/packages/cli/src/commands/_shared/scene-build.test.ts
+++ b/packages/cli/src/commands/_shared/scene-build.test.ts
@@ -9,7 +9,7 @@ import { mkdirSync, mkdtempSync, writeFileSync, readFileSync, rmSync } from "nod
 import { tmpdir } from "node:os";
 import { join } from "node:path";
 
-import { executeSceneBuild } from "./scene-build.js";
+import { executeSceneBuild, type BuildReport } from "./scene-build.js";
 import { buildEmptyRootHtml } from "./scene-project.js";
 
 // ── Module mocks (must be hoisted before the imported module loads) ─────
@@ -232,12 +232,12 @@ afterEach(() => {
   delete process.env.VIBE_BUILD_MODE;
 });
 
-function readBuildReport(): Record<string, any> {
+function readBuildReport(): BuildReport {
   return JSON.parse(readFileSync(join(projectDir, "build-report.json"), "utf-8"));
 }
 
 function expectBuildReportContract(
-  report: Record<string, any>,
+  report: BuildReport,
   expected: Record<string, unknown> = {}
 ): void {
   expect(report).toMatchObject({
@@ -557,7 +557,7 @@ backdrop: "../outside.png"
       selectedStage: "sync",
       success: true,
     });
-    expect(report.stageReports.render.status).toBe("skipped");
+    expect(report.stageReports!.render.status).toBe("skipped");
   });
 
   it("writes a stable beat-only assets build-report contract", async () => {
@@ -741,7 +741,7 @@ backdrop: "This should not dispatch."
       currentStage: "assets",
     });
     expect(report.code).toBe("STORYBOARD_VALIDATION_FAILED");
-    expect(report.validation.ok).toBe(false);
+    expect(report.validation!.ok).toBe(false);
     expect(report.retryWith).toEqual(r.retryWith);
   });
 

--- a/packages/cli/src/commands/_shared/scene-build.ts
+++ b/packages/cli/src/commands/_shared/scene-build.ts
@@ -2118,7 +2118,9 @@ async function finalizeBuildResult(
   return withMeta;
 }
 
-function toBuildReport(projectDir: string, result: SceneBuildResult): Record<string, unknown> {
+export type BuildReport = ReturnType<typeof toBuildReport>;
+
+function toBuildReport(projectDir: string, result: SceneBuildResult) {
   let beatCursor = 0;
   return {
     schemaVersion: "1",

